### DISCUSSION TITLE
Access File in Event Record Hooks

### DISCUSSION THREAD

**Discussion by iamdineshkumar** [pocketbase/pocketbase#4586] [Q&A] [answered]
I need to integrate a third party API which requires image to be passed as base64 encoded value in post request along with other details.  I have files in template collection. 
Template Collection Fields
content Text Field
file Single File field which accepts image/jpeg or image/png

Now i need to create base64 string (eg. ```data:image/jpeg;base64,iVBORw0``` format) for the file in template collection but there is no get method in it record model for files

**Comment by ganigeorgiev** [comment, marked as answer]
The record model stores only the name of the file.

To read the file you'll have to call [filesystem/System.GetFile](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.22.4/tools/filesystem#System.GetFile) with the full filename key.

Here is one possible Go example (_I haven't tested it, treat it as pseudo-code_):
```go
// construct the full file key (usually in the format :collectionId/:recordId/:filename)
key := path.Join(record.BaseFilesPath(), record.GetString("file"))

// initialize the app filesystem (S3 or local)
fsys, err := app.NewFilesystem()
if err != nil {
    return err
}
defer fsys.Close()

// fetch the file from the filesystem (S3 or local)
br, err := fsys.GetFile(key)
if err != nil {
    return err
}
defer br.Close()

// read the entire file content into the bytes buffer
var buf bytes.Buffer
if _, err := br.WriteTo(&buf); err != nil {
    return err
}

// convert to base64 string
// 
// note that you may have to add the "data:image/png;base64," prefix;
// (see https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URLs)
result := base64.StdEncoding.EncodeToString(buf.Bytes())
```

For the JS `pb_hooks` the process is similar but the difficult part will be the last step as we don't have base64 helpers exposed, nor support for Node apis like `Buffer` and you may have to [load an external library](https://pocketbase.io/docs/js-overview/#loading-modules) or call a shell command using the [`$os` helpers](https://pocketbase.io/jsvm/functions/_os.cmd.html) to convert the raw bytes into base64 string.

**Reply by iamdineshkumar** [reply]
I am working in go lang. So this works perfect for me.
